app/views/return-requirements/check.njk

```diff
@@ -113,13 +113,160 @@
         {# Set an easier to use index #}
         {% set rowIndex = loop.index0 %}
         <div>
-          <h3 class="govuk-body">{{requirement.siteDescription}}</h3>
-          {{ govukButton({
-            text: "Remove requirement",
-            classes: "govuk-button--warning",
-            preventDoubleClick: true,
-            href: "/system/return-requirements/" + sessionId + "/remove/" + rowIndex
-          }) }}
+            {{ govukSummaryList({
+              card: {
+                title: {
+                  text: requirement.siteDescription
+                }
+              },
+              classes: "govuk-summary-list--no-border",
+              rows: [
+                {
+                  key: {
+                    text: "Purpose"
+                  },
+                  value: {
+                    html: "requirement.purposes"
+                  },
+                  actions: {
+                    items: [
+                      {
+                        href: "purpose/" + rowIndex,
```

Comment:
We can't use the row index as the requirement index here. When @rvsiyad looked at adding an additional requirement, he found it was perfectly possible for a user to start the journey, but then back all the way out without completing it.

That is why we have the concept of creating an array of `completedRequirements` in the presenter.

So, imagine you have create the first, started adding a second but then backed out (maybe you were unsure of something). Having confirmed your missing detail you then start adding a third which you complete.

In this page you'll just see 2 requirements listed. Left as is, when I click the 'Change' links for the second _display_ requirement, `/points/1` for example, I'll be shown the details for the second incomplete one when I follow it.

Or (!!) we could. But we'd need to add something to `CheckService` that clears out incomplete requirements from `session.requirements`, then save the update to the session. You could then drop leave this view as is and drop the checks for completed requirements in the presenter.

🤷 I'm fine either way!

Reply:
@Cruikshanks amended, works as expected